src/ccl_massfunc.c

```diff
@@ -574,7 +574,7 @@ double ccl_massfunc(ccl_cosmology *cosmo, double halomass, double a, double odel
 	  return NAN;
   }
 
-  if (halomass > 1e17 || halomass < 1e6){
+  if (log10(halomass) > cosmo->spline_params.LOGM_SPLINE_MAX || log10(halomass) < cosmo->spline_params.LOGM_SPLINE_MIN){
```

Comment:
Could we take log10(halomass) above and assign it to a variable to avoid having to calculate it twice?

Reply:
Done!